### ISSUE TITLE
Parsing f-strings with seeming assignment expressions produces incorrect SyntaxError

### ISSUE THREAD

**Issue by gousaiyang** [davidhalter/parso#87]
There is a subtle issue when parsing f-strings whose expression parts look like an assignment expression in Python 3.8. As illustrated in [PEP 572](https://www.python.org/dev/peps/pep-0572/#exceptional-cases), both `f'{(x:=10)}'` and `f'{x:=10}'` are valid expressions.

```
>>> f'{(x:=10)}'  # Valid, uses assignment expression
'10'
>>> x = 10
>>> f'{x:=10}'    # Valid, passes '=10' to formatter
'        10'
```

The former is an f-string with an assignment expression, while the latter is an f-string with `=10` as the format specifiers of `x`. Both are valid, although this may be confusing and is not a good programming practice.

However, parso fails to parse `f'{x:=10}'`:

```
>>> import parso
>>> parso.parse("f'{x:=10}'", error_recovery=False)
Traceback (most recent call last):
  File "C:\Python38\lib\site-packages\parso\parser.py", line 181, in _add_token
    plan = stack[-1].dfa.transitions[transition]
KeyError: ReservedString(:=)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Python38\lib\site-packages\parso\__init__.py", line 58, in parse
    return grammar.parse(code, **kwargs)
  File "C:\Python38\lib\site-packages\parso\grammar.py", line 78, in parse
    return self._parse(code=code, **kwargs)
  File "C:\Python38\lib\site-packages\parso\grammar.py", line 147, in _parse
    root_node = p.parse(tokens=tokens)
  File "C:\Python38\lib\site-packages\parso\python\parser.py", line 82, in parse
    return super(Parser, self).parse(tokens)
  File "C:\Python38\lib\site-packages\parso\parser.py", line 128, in parse
    self._add_token(token)
  File "C:\Python38\lib\site-packages\parso\parser.py", line 187, in _add_token
    self.error_recovery(token)
  File "C:\Python38\lib\site-packages\parso\python\parser.py", line 151, in error_recovery
    return super(Parser, self).error_recovery(token)
  File "C:\Python38\lib\site-packages\parso\parser.py", line 151, in error_recovery
    raise ParserSyntaxError('SyntaxError: invalid syntax', error_leaf)
parso.parser.ParserSyntaxError: ('SyntaxError: invalid syntax', <ErrorLeaf: TokenType(OP):':=', (1, 4)>)
```

I am using parso 0.5.1. This issue only happens on Python 3.8.

**Comment by davidhalter** [comment]
This is definitely a but with Python 3.8.

**Comment by davidhalter** [comment]
This was also fixed by #93

**Comment by gousaiyang** [comment]
Thanks for fixing this!
